Minio/V4Authenticator.cs

```diff
@@ -383,22 +384,22 @@ internal string GetPresignCanonicalRequest(IRestClient client, IRestRequest requ
             // QUERY
             if (path.Length == 2)
             {
-                var parameterString = path[1];
-                var parameterList = parameterString.Split('&');
-                var sortedQueries = new SortedSet<string>();
-                foreach (string individualParameterString in parameterList)
+                var queryParams = path[1].Split('&').Select(p => p.Split('='))
```

Comment:
This change is not related to Select right?

Reply:
Query parameters were not being canonicalized correctly - a preexisting bug exposed while implementing the select API. Hence this is a part of the PR.